README.md

```diff
@@ -1,15 +1,117 @@
 
-TBD
+- **Terraform Installed**: You should have Terraform installed on your local machine. You can download Terraform from the [official website](https://www.terraform.io/downloads.html).
 
-## License
+- **AWS Credentials**: Ensure that you have AWS credentials configured on your machine. You can configure AWS access keys and secret keys using the [AWS CLI](https://aws.amazon.com/cli/).
 
-This code is released under the Apache 2.0 License. See [LICENSE](LICENSE) for more details.
+- **Terraform AWS Provider Configuration**: Configure the AWS provider in your Terraform project. You can add your AWS access and secret keys as environment variables or use other methods for provider configuration.
 
-## Contributing
+- **Terraform Configuration File**: Create a new Terraform configuration file in your project, and use this module by specifying its source.
+
+## Usage
+
+To use this module, follow these steps:
+
+Copy and paste into your Terraform configuration, insert the variables, and run ``terraform init``:
+
+```hcl
+module "graphdb" {
+  source  = "Ontotext-AD/graphdb/aws"
+  version = "0.0.1"
+}
+```
+
+## Configuration
+
+This README provides more detailed information about the important variables and how to configure them for the Terraform AWS GraphDB module. Please ensure that you adapt the configuration to match your specific use case and requirements.
+
+### Important Variables (Inputs)
+
+The following are the important variables you should configure when using this module:
+
+- `aws_region`: The region in which GraphDB will be deployed.
+
+- `vpc_id`: The ID of the Virtual Private Cloud (VPC) where you want to create the AWS GraphDB cluster. This is a required variable.
+
+- `instance_type`: The instance type for the GDB cluster nodes. This should match your performance and cost requirements.
+
+- `cluster_size`: The number of instances in the cluster. Recommended is 3, 5 or 7 in order to have consensus according to the [RAFT algorithm](https://raft.github.io/). 
+
+- `subnet_ids`: A list of subnet IDs in your VPC to place the Neptune instances. Ensure that these subnets have proper route tables and security groups configured.
+
+- `security_group_ids`: A list of security group IDs to associate with the instances for network security.
+
+
+
+## Examples
+
+This repository includes usage examples in the `examples/` directory, such as:
+
+- [VPC with Multiple Availability Zones](./examples/vpc-with-multiple-az): Demonstrates creating a VPC with multiple Availability Zones and provisioning an AWS GraphDB cluster.
+
+Here's an example of how to configure the module in your Terraform project:
+
+#### Example Configuration
+
+##### terraform.tfvars
+```hcl
 
+aws_region = "us-east-1"
+
+azs = ["us-east-1a", "us-east-1b", "us-east-1c"]
+
+resource_name_prefix = "my-prefix"
+
+graphdb_license_path = "/path/to/your/license.license"
+
+tags = {
+  Name = "MyInstance"
+  Environment = "Production"
+}
+
+ami_id = "GraphDB-AMI"
```

Comment:
@mihailradkov What are we going to do with this? Users cannot use our AMI - should we add instructions on how to build one?

Reply:
this will be addressed separately